NYPL-discovery/discovery-front-end
# Scc 2608 bugfix

**What's this do?**
Fixes a bug in the way we detect redirect loops. Instead of counting redirects when the account page loads (the end of the loop) we count redirects when we send the patron to the login page (the beginning of the loop). The previous implementation had a problem with false positives, because there are many other things that can cause the page to reload in the given timeframe. This should fix the problem since sending the patron to the login page is always a redirect. 

**Why are we doing this? (w/ JIRA link if applicable)**
This is 2608

**Do these changes have automated tests?**
Fixed the existing automated tests to match the new implementation

**How should this be QAed?**
Reproducing the loop: modify the line

`'src', 'https://browse.nypl.org/iii/encore/logoutFilterRedirect?suite=def'`

in `logoutUtils.js` to something else. I have been using

`'src', 'https://browse.nypl.org'`

This simulates an error in the logout page.

Navigate to the account page and log in. Then delete cookies (this simulates` III_SESSION` not being set properly), and reload. You should see the redirect loop, and eventually be redirected to a 404 page.

When not simulating the redirect loop, everything should work as before.

**Dependencies for merging? Releasing to production?**
NA

**Has the application documentation been updated for these changes?**
NA

**Did someone actually run this code to verify it works?**
I tested locally.
